doc/sphinx/proof-engine/ltac.rst

```diff
@@ -29,159 +30,99 @@ Syntax
-                     : `int`
-                     : ( `ltac_expr` )
-   component : `string` | `qualid`
-   message_token     : `string` | `ident` | `int`
-   tacarg            : `qualid`
-                     : ()
-                     : ltac : `atom`
-                     : `term`
-   let_clause        : `ident` [`name` ... `name`] := `ltac_expr`
-   context_rule      : `context_hyp`, ..., `context_hyp` |- `cpattern` => `ltac_expr`
-                     : `cpattern` => `ltac_expr`
-                     : |- `cpattern` => `ltac_expr`
-                     : _ => `ltac_expr`
-   context_hyp       : `name` : `cpattern`
-                     : `name` := `cpattern` [: `cpattern`]
-   match_rule        : `cpattern` => `ltac_expr`
-                     : context [`ident`] [ `cpattern` ] => `ltac_expr`
-                     : _ => `ltac_expr`
-   test              : `int` = `int`
-                     : `int` (< | <= | > | >=) `int`
-   selector          : [`ident`]
-                     : `int`
-                     : (`int` | `int` - `int`), ..., (`int` | `int` - `int`)
-   toplevel_selector : `selector`
-                     : all
-                     : par
-                     : !
-
-.. productionlist:: coq
-   top              : [Local] Ltac `ltac_def` with ... with `ltac_def`
-   ltac_def         : `ident` [`ident` ... `ident`] := `ltac_expr`
-                    : `qualid` [`ident` ... `ident`] ::= `ltac_expr`
+In Coq, the expression :n:`try repeat @tactic__1 || @tactic__2; @tactic__3; @tactic__4`
+is interpreted as :n:`(try (repeat (@tactic__1 || @tactic__2)); @tactic__3); @tactic__4`
+because `||` is part of :token:`ltac_expr2`, which has higher precedence than
+:tacn:`try` and :tacn:`repeat` (at the level of :token:`ltac_expr3`), which
+in turn have higher precedence than `;`, which is part of :token:`ltac_expr4`.
+(A *lower* number in the nonterminal name means *higher* precedence in this grammar.)
 
-Tactics in terms
-~~~~~~~~~~~~~~~~
+The constructs in :token:`ltac_expr` are :term:`left associative`.
 
-.. insertprodn term_ltac term_ltac
+.. insertprodn ltac_expr tactic_atom
 
 .. prodn::
-   term_ltac ::= ltac : ( @ltac_expr )
+   ltac_expr ::= {| @ltac_expr4 | @binder_tactic }
+   ltac_expr4 ::= @ltac_expr3 ; {| @ltac_expr3 | @binder_tactic }
+   | @ltac_expr3 ; [ @for_each_goal ]
+   | @ltac_expr3
+   ltac_expr3 ::= @l3_tactic
+   | @ltac_expr2
+   ltac_expr2 ::= @ltac_expr1 + {| @ltac_expr2 | @binder_tactic }
+   | @ltac_expr1 || {| @ltac_expr2 | @binder_tactic }
+   | @l2_tactic
+   | @ltac_expr1
+   ltac_expr1 ::= @tactic_value
+   | @qualid {* @tactic_arg }
+   | @l1_tactic
+   | @ltac_expr0
+   tactic_value ::= {| @value_tactic | @syn_value }
+   tactic_arg ::= @tactic_value
+   | @term
+   | ()
+   ltac_expr0 ::= ( @ltac_expr )
+   | [> @for_each_goal ]
+   | @tactic_atom
+   tactic_atom ::= @int
+   | @qualid
+   | ()
+
+.. todo PR For the moment, I've left the language constructs like +, || and ; unchanged in the grammar.
```

Comment:
I do not think that this has to be resolved before merging this PR. I'm fine with the present state.